ims-apple/IMS Apple/Views/AddingProduct/AddingProductView.swift

```diff
@@ -0,0 +1,172 @@
+            Text("Estado")
+                .frame(maxWidth: .infinity, alignment: .leading)
+            
+            Menu {
+                ForEach(["Disponible", "Ocupado", "Agotado"], id: \.self) { item in
+                    Button(item) {
+                        withAnimation {
+                            selectedState = item
+                        }
+                    }
+                }
+            } label: {
+                Text(selectedState)
+            }
+            .menuStyle(.borderlessButton)
+            .padding(.horizontal)
+            .frame(maxWidth: .infinity, minHeight: Constants.minHeight, alignment: .leading)
+            .overlay {
+                RoundedRectangle(cornerRadius: Constants.cornerRadiusSize)
+                    .stroke(.graySecundary)
+            }
+        }
+    }
+    
+    // MARK: - Product Detail
+    
+    private var productDetailsView: some View {
+        VStack(alignment: .leading, spacing: Constants.horizontalSpacing) {
+            IMSTextField(
+                type: .custom("Detalles del producto"),
+                text: $productDetails,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity,
+                minHeight: Constants.detailButtonHeight
+            )
+            .padding(.vertical)
+            
+            photoPickerView
+        }
+    }
+    
+    // MARK: - Photo Picker
+    
+    private var photoPickerView: some View {
+        VStack(spacing: Constants.spaceSize) {
+            if let avatarImage {
+                avatarImage
+                    .resizable()
+                    .frame(maxWidth: .infinity, maxHeight: Constants.photoMaxHeight)
+            } else {
+                RoundedRectangle(cornerRadius: Constants.cornerRadiusSize)
+                    .stroke(style: StrokeStyle(dash: [Constants.strokLengths]))
+                    .frame(maxWidth: .infinity, maxHeight: Constants.photoMaxHeight)
+            }
+            
+            HStack {
+                Image(systemName: Constants.loadingImage)
+                
+                PhotosPicker("Cargar Imagen", selection: $avatarItem, matching: .images)
+                    .buttonStyle(.plain)
+            }
+            .padding(.horizontal)
+            .frame(maxWidth: .infinity, maxHeight: Constants.buttonHeight)
+            .background {
+                RoundedRectangle(cornerRadius: Constants.cornerRadiusSize)
+                    .fill(LinearGradient(colors: [.purpleGradient, .blueGradient], 
+                                         startPoint: .leading,
+                                         endPoint: .trailing))
+            }
+        }
+        .onChange(of: avatarItem) {
+            Task {
+                if let loaded = try? await avatarItem?.loadTransferable(type: Image.self) {
+                    avatarImage = loaded
+                } else {
+                    print("Failed")
+                }
+            }
+        }
```

Comment:
I think this code maybe generate a `silent errors`, I modified some the code to avoid the `silent errors`.

```suggestion
        onChange(of: avatarItem) {
            Task {
                do {
                    if let imageLoaded = try await avatarItem?.loadTransferable(type: Image.self) {
                        avatarImage = imageLoaded
                    } else {
                        debugPrint("Avatar item is currently nil") // We need to avoid the silent errors
                    }
                } catch {
                    debugPrint("Failed loading the image with error: \(error.localizedDescription)")
                }
            }
        }
```

Reply:
Add to the viewMOdel